sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -762,4 +762,15 @@ object QueryPlan extends PredicateHelper {
       case e: AnalysisException => append(e.toString)
     }
   }
+
+  /**
+   * Generate detailed field string with different format based on type of input value
```

Comment:
this comment and method name are a bit generic; could we expand them to mention what type of field we are referring to here, and when this might be used? can we give a brief example?